profiles/coreos/base/profile.bashrc.slsa-provenance

```diff
@@ -0,0 +1,182 @@
+    csum="$(cat "/mnt/host/source/src/scripts/.git/modules/sdk_container/src/third_party/${repo}/ORIG_HEAD")"
+    uri="git+https://github.com/flatcar-linux/${repo}.git/${ebuild}"
+    echo -e "      { \"uri\": \"${uri}\","
+    echo -n "        \"digest\": {\"sha1\":\"${csum}\"} }"
+
+    # The main sources
+    if [ -n "${A}" ] ; then
+        # Package is built from downloaded source tarball(s)
+        # There can be multiple, and can be used conditionally based on use flags,
+        #  and even replaced with different local names ("http://... -> othername.tgz"). So
+        #  we go through what's actually used ($A), then find the corresponding source URI.
+        local src="" prev_uri="" rename="false" orig_name=""
+        for src in ${A}; do
+            local found="false"
+            for uri in ${SRC_URI}; do
+                if [ "${uri}" = "->" ] ; then
+                    rename="true"
+                    continue
+                fi
+                if [ "${src}" = "$(basename "${uri}")" ] ; then
+                    orig_name="${src}"
+                    if [ "${rename}" = "true" ] ; then
+                        uri="${prev_uri}"
+                        orig_name="$(basename "${uri}")"
+                    fi
+                    einfo "    Provenance: recording tarball material (input) '${src}' ('${orig_name}')"
+                    csum="$(sha512sum "${DISTDIR}/${src}" | cut -d' ' -f1)"
+                    echo -e ",\n      { \"uri\": \"${uri}\","
+                    echo -n "        \"digest\": {\"sha512\":\"${csum}\"} }"
+                    found="true"
+                fi
+                rename="false"
+                prev_uri="${uri}"
+            done
+            if [ "${found}" != "true" ] ; then
+                die "No SRC_URI found for source '${src}', unable to record provenance!"
+            fi
+        done
+    elif [ -n "${EGIT_REPO_URI:-}" ] ; then
+        # package is built from repo checkout (git)
+        einfo "    Provenance: recording GIT material (input) '${EGIT_REPO_URI}'"
+        csum="${EGIT_COMMIT}"
+        uri="${EGIT_REPO_URI}"
+        echo -e ",\n      { \"uri\": \"${uri}\","
+        echo -n "        \"digest\": {\"sha1\":\"$csum\"} }"
+    fi
+
+    # Patches / files shipped with the ebuild (if any)
+    uri="git+https://github.com/flatcar-linux/${repo}.git/${CATEGORY}/${PN}/files"
+    csum="$(cat "/mnt/host/source/src/scripts/.git/modules/sdk_container/src/third_party/${repo}/ORIG_HEAD")"
+    if [ -d "${FILESDIR}" ] ; then
+        for file in $(cd "$FILESDIR" && find . -type f | sed 's:^./::') ; do
+            einfo "    Provenance: recording ebuild material (input) '${file}'"
+            echo -e ",\n      { \"uri\": \"${uri}/${file}\","
+            echo -n "        \"digest\": {\"sha1\":\"$csum\"} }"
+        done
+    fi
+
+    echo -ne '\n   ]'
+}
+# --
+
+__slsa_provenance_report() {
+    local scripts_hash="$(cat "/mnt/host/source/src/scripts/.git/ORIG_HEAD")"
```

Comment:
This probably should use HEAD, but this maybe in form of `ref: refs/heads/main`, in such case a value from `.git/refs/heads/main` should be taken.

Reply:
this uses HEAD now and resolved the ref